### ISSUE TITLE
 MultipleNullnessAnnotations inconsistent on nested class

### ISSUE THREAD

**Issue by xenoterracide** [google/error-prone#4320]
supressing the warning on the nested static class fixes the error, no suppression is needed on the outer class. This is inconsistent and makes no sense to me.

```
/home/xeno/IdeaProjects/spring-app-commons/module/jpa/src/main/java/com/xenoterracide/jpa/AbstractUuidEntityBase.java:112: warning: [MultipleNullnessAnnotations] This type use has conflicting nullness annotations
```

```java
package com.xenoterracide.jpa;

import jakarta.validation.constraints.NotNull;
import java.util.UUID;
import org.jspecify.annotations.NonNull;
import org.jspecify.annotations.Nullable;

public abstract class AbstractUuidEntityBase<ID extends AbstractUuidEntityBase.AbstractIdentity> {

  @NotNull
  private @Nullable ID id;

  protected AbstractUuidEntityBase(@NonNull ID id) {
    this.id = id;
  }

  public abstract static class AbstractIdentity {

    @NotNull
    private @Nullable UUID id;

    protected AbstractIdentity(@NonNull UUID id) {
      this.id = id;
    }
  }
}
```

**Comment by cushon** [comment]
Suppressions should be handled on any syntactically enclosing scope, i.e. I'd expect any of the following to suppress the diagnostic. Can you elaborate on how the behaviour you're seeing is inconsistent / surprising?

```java
class AbstractUuidEntityBase {
  class AbstractIdentity {
    @SuppressWarnings("MultipleNullnessAnnotations")
    @NotNull @Nullable UUID id; // This type use has conflicting nullness annotations
  }
}
```

```java
class AbstractUuidEntityBase {
  @SuppressWarnings("MultipleNullnessAnnotations")
  class AbstractIdentity {
    @NotNull
    private @Nullable UUID id; // This type use has conflicting nullness annotations
  }
}
```

```java
@SuppressWarnings("MultipleNullnessAnnotations")
class AbstractUuidEntityBase {
  class AbstractIdentity {
    @NotNull @Nullable UUID id; // This type use has conflicting nullness annotations
  }
}
```

**Comment by xenoterracide** [comment]
The problem is that in my code I only needed an annotation on the nested field, I didn't need it on the outer fields. I only got the error once.

**Comment by cushon** [comment]
I am not sure I follow. Can you share a minimal example that reproduces the issue you're experiencing?

**Comment by xenoterracide** [comment]
Sorry, I realized I copied the wrong class. I updated the original example. It only generates 1 error, it should generate 2. This will fix it.

```java
package com.xenoterracide.jpa;

import jakarta.validation.constraints.NotNull;
import java.util.UUID;
import org.jspecify.annotations.NonNull;
import org.jspecify.annotations.Nullable;

public abstract class AbstractUuidEntityBase<ID extends AbstractUuidEntityBase.AbstractIdentity> {

  @NotNull
  private @Nullable ID id;

  protected AbstractUuidEntityBase(@NonNull ID id) {
    this.id = id;
  }

  public abstract static class AbstractIdentity {

    @NotNull
    @SuppressWarnings("MultipleNullnessAnnotations")
    private @Nullable UUID id;

    protected AbstractIdentity(@NonNull UUID id) {
      this.id = id;
    }
  }
}
```

**Comment by cushon** [comment]
Thanks. I think there is an issue with the annotation on the type parameter use of `ID` not being visible to Error Prone.

Repro:

```java
import jakarta.validation.constraints.NotNull;
import org.jspecify.annotations.Nullable;

class T<X> {
  @NotNull @Nullable X one;
  @NotNull @Nullable int two;
}
```

```
javac -J--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED   -J--add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED   -J--add-exports=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED   -J--add-exports=jdk.compiler/com.sun.tools.javac.model=ALL-UNNAMED   -J--add-exports=jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED   -J--add-exports=jdk.compiler/com.sun.tools.javac.processing=ALL-UNNAMED   -J--add-exports=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED   -J--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED   -J--add-opens=jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED   -J--add-opens=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED   -XDcompilePolicy=simple   -processorpath error_prone_core-${EP_VERSION?}-with-dependencies.jar:dataflow-errorprone-${DATAFLOW_VERSION?}.jar -cp jakarta.validation-api-3.1.0-M1.jar:jspecify-0.3.0.jar   '-Xplugin:ErrorProne -XepDisableAllChecks -Xep:MultipleNullnessAnnotations:ERROR'   T.java
```

Expected: findings on both `one` and `two`.

Actual: the finding shows up only on `two`:

```
T.java:8: error: [MultipleNullnessAnnotations] This type use has conflicting nullness annotations
  @NotNull @Nullable int two;
                         ^
    (see https://errorprone.info/bugpattern/MultipleNullnessAnnotations)
1 error
```
